tls/s2n_recv.c

```diff
@@ -110,6 +110,8 @@ int s2n_read_full_record(struct s2n_connection *conn, uint8_t *record_type, int
 
 ssize_t s2n_recv_impl(struct s2n_connection *conn, void *buf, ssize_t size, s2n_blocked_status *blocked)
 {
+    POSIX_ENSURE_GTE(size, 0);
+    size_t validated_size = size;
```

Comment:
Same feedback as for s2n_send_early_data_impl: give the variable you're going to use throughout the method the better name. The `size_t` should be `size`, the `ssize_t` should be something else.